src/App.tsx

```diff
@@ -1,14 +1,54 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { Options } from './types/Options';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [chosenUserId, setChosenUserId] = useState<number>(0);
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+  const [isShown, setIsShown] = useState(false);
+  const [isModalLoading, setIsModalLoading] = useState(false);
+  const [option, setOption] = useState<Options>(Options.All);
+  const [query, setQuery] = useState('');
```

Comment:
```suggestion
  const [todos, setTodos] = useState<Todo[]>([]);
  const [todo, setTodo] = useState<Todo | null>(null);
  const [option, setOption] = useState<Options>(Options.All);
  const [query, setQuery] = useState('');
```
it is enough states for App component.

user state can be in TodoModal component. no need to know about user in App. All what we know it is userId in todo.

maybe also loading for todos can make, but it is enough in this task to check todo.length and show or not Loader based on it.